baseplate/experiments/providers/simple_experiment.py

```diff
@@ -0,0 +1,150 @@
+
+
+class SimpleExperiment(Experiment):
+    """Base class for simple experiments. Note that this class shares a lot of
+    code from the r2.py class, which is slated for deprecation and removal.
+    """
+
+    def make_seed(self, seed_data):
+        id = seed_data.get('id')
+        name = seed_data.get('name')
+        version = seed_data.get('shuffle_version')
+
+        return "{}.{}.{}".format(id, name, version)
+
+    def __init__(self, id, name, owner, start_ts, stop_ts, config,
+                 enabled=True, **kwargs):
+
+        self.id = id
+        self.name = name
+        self.owner = owner
+        self.num_buckets = 1000
+
+        self.start_ts = start_ts
+        self.stop_ts = stop_ts
+        self.enabled = enabled
+
+        self.bucket_val = config.get("bucket_val", "user_id")
+        self.version = config.get("experiment_version")
+        self.shuffle_version = config.get("shuffle_version")
+
+        self.variants = config.get("variants", [])
+
+        seed_data = {"id": id, "name": name, "shuffle_version": self.shuffle_version}
+        self.seed = config.get("bucket_seed", self.make_seed(seed_data))
+
+        self._validate_variants(self.variants)
+
+    def _get_seed(self):
+        return self.seed
+
+    def get_unique_id(self, **kwargs):
+        if kwargs.get(self.bucket_val):
+            return ":".join(
+                [self.name, self.bucket_val, str(kwargs[self.bucket_val])]
+            )
+        else:
+            return None
+
+    def should_log_bucketing(self):
+        """ Default to true. Override if logging of eligibility events not required. """
+        return True
+
+    def variant(self, **kwargs):
+        lower_kwargs = {k.lower(): v for k, v in iteritems(kwargs)}
+
+        if not self._is_enabled():
+            return None
+
+        variant = self._check_overrides(**lower_kwargs)
+        if variant is not None and variant in self.variants:
+            return variant
+
+        if self.bucket_val not in lower_kwargs:
+            logger.info(
+                "Must specify %s in call to variant for experiment %s.",
+                self.bucket_val,
+                self.name,
+            )
+            return None
+        if lower_kwargs[self.bucket_val] is None:
+            logger.info(
+                "Cannot choose a variant for bucket value %s = %s "
+                "for experiment %s.",
+                self.bucket_val,
+                lower_kwargs[self.bucket_val],
+                self.name,
+            )
+            return None
+
+        if not self._is_targeted(**lower_kwargs):
+            return None
+
+        bucket = self._calculate_bucket(lower_kwargs[self.bucket_val])
+        return self._choose_variant(bucket)
+
+    def _check_overrides(self, **kwargs):
+        """Check if any of the kwargs override the variant. Functionality
+        to be built in the future. For now, overrides are not supported.
+        """
+        return None
```

Comment:
IMO, we should leave code like this out if it's not being implemented, we can implement it when we need it rather than having code that doesn't actually do anything lying around until we do.

Reply:
Normally, I'd agree. In this case, however, I think there's value in including it for 2 reasons:
1) to indicate that this code is coming shortly (it's currently in-progress)
2) by including the non-functionality here, anyone using this can easily override targeting and overrides in subclasses.

Reply:
"coming shortly" is weird in fixed-release libraries though.

Reply:
2 to 1. Removed.